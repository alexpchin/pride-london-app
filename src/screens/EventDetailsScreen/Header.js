// @flow
import React from "react";
import {
  Text,
  StyleSheet,
  TouchableOpacity,
  StatusBar,
  ImageBackground
} from "react-native";
import SafeAreaView from "react-native-safe-area-view";

type Props = {
  onBackButtonPress: () => void,
  imageUrl: string
};

const Header = ({ onBackButtonPress, imageUrl }: Props) => (
  <ImageBackground
    style={styles.header}
    source={{ uri: imageUrl }}
    resizeMode="cover"
  >
    <SafeAreaView style={styles.content}>
      <StatusBar barStyle="default" animated />
      <TouchableOpacity onPress={onBackButtonPress}>
        <Text>Back</Text>
      </TouchableOpacity>
    </SafeAreaView>
  </ImageBackground>
);

const styles = StyleSheet.create({
  header: {
    flex: 1,
    height: 180
  },
  content: {
    padding: 22
  }
});

export default Header;
