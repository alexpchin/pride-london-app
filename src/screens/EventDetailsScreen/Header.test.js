// @flow
import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

it("renders correctly", () => {
  const output = shallow(
    <Header onBackButtonPress={() => {}} imageUrl="https://image.jpg" />
  );
  expect(output).toMatchSnapshot();
});

it("calls callback when pressed", () => {
  const onPress = jest.fn();
  const output = shallow(
    <Header onBackButtonPress={onPress} imageUrl="https://image.jpg" />
  );

  const touchable = output.find("TouchableOpacity");
  touchable.simulate("press");

  expect(onPress).toHaveBeenCalled();
});
