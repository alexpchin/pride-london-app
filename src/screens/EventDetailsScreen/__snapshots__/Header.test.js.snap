// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageBackground
  resizeMode="cover"
  source={
    Object {
      "uri": "https://image.jpg",
    }
  }
  style={
    Object {
      "flex": 1,
      "height": 180,
    }
  }
>
  <withOrientation
    style={
      Object {
        "padding": 22,
      }
    }
  >
    <StatusBar
      animated={true}
      barStyle="default"
      showHideTransition="fade"
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Back
      </Text>
    </TouchableOpacity>
  </withOrientation>
</ImageBackground>
`;
